src/lib/cache.coffee

```diff
@@ -66,17 +66,17 @@ module.exports = (dnschain) ->
                     @cache.setex(key, ttl, JSON.stringify value) if not err2
                     valueDoer err, key, value
 
-        resolveBlockchain: (resolver, path, options, cb) ->
+        resolveResource: (resolver, resourceFn, serialization, args, cb) ->
             if @blockchainEnabled and resolver.cacheTTL?
                 retriever = (key, callback) =>
                     f = (err, result) =>
                         callback err, resolver.cacheTTL, result
-                    resolver.resolve path, options, f
+                    resourceFn.call resolver, args..., f
                 doer = (err, key, result) =>
                     cb err, result
-                @get "#{resolver.name}:#{path}:#{JSON.stringify(options)}", retriever, doer
+                @get serialization, retriever, doer
             else
-                resolver.resolve path, options, cb
+                resourceFn.call resolver, args..., cb
```

Comment:
The `args` thing is confusing me.

In all cases in this function you are simply passing them to another parameter... so couldn't you get rid of the `args` parameter and use them directly within the `resourceFn` that you're passing in?